### PR TITLE
Fix lint issues

### PR DESCRIPTION
**Pull request by marcoshuck**
This PR fixes the issues found in the lint job

### CODE REVIEW

File: bundles/worlds/worlds_service.go [hunk 1]
```diff
@@ -694,10 +694,18 @@ func parseModelIncludes(tx *gorm.DB, world *World,
...
 	}
-	defer xmlFile.Close()
-	b, _ := ioutil.ReadAll(xmlFile)
+	defer func(file *os.File) {
+		err := file.Close()
+		if err != nil {
+			log.Println("Failed to close file:", err)
+		}
+	}(xmlFile)
+	b, _ := io.ReadAll(xmlFile)
```
Comment (marcoshuck) [inline bundles/worlds/worlds_service.go]: This should be checked

File: README.md [hunk 2]
```diff
@@ -378,7 +378,7 @@ run (from the proto folder):
     protoc --go_out=. *.proto
     ```
 
-    Then update the generated import proto from code.google... to "github.com/golang/protobuf/proto"
+    Then update the generated import proto from code.google... to "google.golang.org/protobuf/proto"
```
Comment (AlejoAsd) [inline README.md:381]: Not for this PR, but I wonder if we still have to manually change this with the latest versions of protobuf.

File: bundles/worlds/worlds_service.go [hunk 3]
```diff
@@ -694,10 +694,18 @@ func parseModelIncludes(tx *gorm.DB, world *World,
...
 		return nil, gz.NewErrorMessageWithBase(gz.ErrorFormInvalidValue, err)
 	}
-	defer xmlFile.Close()
-	b, _ := ioutil.ReadAll(xmlFile)
+	defer func(file *os.File) {
+		err := file.Close()
+		if err != nil {
+			log.Println("Failed to close file:", err)
+		}
+	}(xmlFile)
```
Comment (AlejoAsd) [inline bundles/worlds/worlds_service.go:702]: Was this added to appease the linter? I'm surprised this is the first time we have had to handle file closing this way.

If it was done to appease the linter, then we will probably have to do this with all our file closing operations across all our projects and this particular function should probably live in `gz-go` instead of being defined here. Also, we should use the [`io.Closer`](https://pkg.go.dev/io#Closer) interface as parameter to make this widely compatible with close operations (e.g. for files, SQL cursors, etc.).

```go
// We can optionally add a string parameter to make the print message specify what failed to close
func Close(closer io.Closer) {
    err := closer.Close()
    if err != nil {
        log.Println("Failed to close:", err)
    }
}
```

File: database.go [hunk 4]
```diff
@@ -435,4 +450,7 @@ func DBPopulate(ctx context.Context, path string, db *gorm.DB, onlyWhenEmpty boo
 			}
 			return nil
 		})
+	if err != nil {
+		return
+	}
```
Comment (AlejoAsd) [inline database.go:455]: This is functionally identical to not having an error handler. Is this causing lint issues? Does it make sense to maybe add `Println` call?

File: handler_collections.go [hunk 5]
```diff
@@ -295,7 +304,7 @@ func collectionAssetAdd(colOwner, colName, assetType string, user *users.User,
 			return nil, em
 		}
 		models.ElasticSearchUpdateModel(r.Context(), tx, *model)
-	} else if assetType == collections.TModel {
+	} else if assetType == collections.TWorld {
```
Comment (AlejoAsd) [inline handler_collections.go:307]: :boot: 
:lady_beetle: 

:clap::clap::clap:

File: proto/buf.gen.go.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Buf is a building tool for generating proto messages and grpc services in different languages.
```
Comment (AlejoAsd) [inline proto/buf.gen.go.yaml:1]: Does this require a README update?

File: vcs/vcs_gogit.go [hunk 7]
```diff
@@ -81,50 +78,6 @@ func (g *GoGitVCS) assertValidRepo() error {
...
-			Name:  gitName,
-			Email: gitEmail,
-			When:  time.Now(),
-		},
-	})
-	if err != nil {
-		return gz.WithStack(err)
-	}
-	return nil
-}
```
Comment (AlejoAsd) [inline vcs/vcs_gogit.go:126]: Why was this removed?

**Review by AlejoAsd — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (marcoshuck) [reply README.md:381]: I don't think so, keep in mind that this documentation is outdated.

File: bundles/worlds/worlds_service.go (hunk 3 above)
Comment (marcoshuck) [reply bundles/worlds/worlds_service.go:702]: This is something that came to my mind while refactoring this, and I'm glad you raise this comment here. I agree we should extract this to `gz-go`, but I can't avoid asking myself: what happens if you wanna introduce a custom error handling action when failing to close a file?

File: proto/buf.gen.go.yaml (hunk 6 above)
Comment (marcoshuck) [reply proto/buf.gen.go.yaml:1]: It does, but maybe we should consider doing something like we did for cloudsim: https://github.com/gazebo-web/cloudsim#readme

and improve the docs using `go doc` guidelines

File: vcs/vcs_gogit.go (hunk 7 above)
Comment (marcoshuck) [reply vcs/vcs_gogit.go:126]: This function was not being used, the linter recommends removing it.

File: bundles/worlds/worlds_service.go (hunk 3 above)
Comment (AlejoAsd) [reply bundles/worlds/worlds_service.go:702]: We can have `Close(closer io.Closer)` and `CloseWithHandler(closer io.Closer, errorHandler func())`.

Depending on the complexity of your handler, it might just make more sense to not use the `gz-go` function and write the special case as an exception.

File: proto/buf.gen.go.yaml (hunk 6 above)
Comment (AlejoAsd) [reply proto/buf.gen.go.yaml:1]: Sounds good. Please create an issue so that we can address this in another PR.

File: vcs/vcs_gogit.go (hunk 7 above)
Comment (AlejoAsd) [reply vcs/vcs_gogit.go:126]: Thanks!
